e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts

```diff
@@ -0,0 +1,211 @@
+
+    expect(await personalFiles.shareDialog.isShareToggleChecked()).toBe(true);
+    await personalFiles.shareDialog.shareToggle.click();
+
+    expect(personalFiles.confirmDialog.getDialogTitle('Remove this shared link')).toBeVisible();
+    expect(
+      personalFiles.confirmDialog.getDialogContent('This link will be deleted and a new link will be created next time this file is shared')
+    ).toBeVisible();
+    expect(personalFiles.confirmDialog.okButton).toBeEnabled();
+    expect(personalFiles.confirmDialog.cancelButton).toBeEnabled();
+  });
+
+  test('[C306996] Unshare a file', async ({ personalFiles, searchPage, nodesApiAction, page }) => {
+    await personalFiles.acaHeader.searchButton.click();
+    await searchPage.searchInput.searchButton.click();
+    await searchPage.searchOverlay.checkFilesAndFolders();
+    await searchPage.searchOverlay.searchFor(file2);
+
+    await personalFiles.dataTable.selectItem(file2);
+    await personalFiles.acaHeader.shareButton.click();
+    const url = await personalFiles.shareDialog.getLinkUrl();
+    await personalFiles.shareDialog.shareToggle.click();
+
+    await personalFiles.confirmDialog.okButton.click();
+    expect(await nodesApiAction.isFileShared(file2Id)).toBe(false);
+
+    await page.goto(url);
+    await personalFiles.viewer.waitForViewerToOpen();
+    expect(await personalFiles.viewer.fileTitleButtonLocator.innerText()).not.toEqual(file2);
+  });
+
+  test('[C306997] Cancel the Unshare action', async ({ personalFiles, searchPage }) => {
+    await personalFiles.acaHeader.searchButton.click();
+    await searchPage.searchInput.searchButton.click();
+    await searchPage.searchOverlay.checkFilesAndFolders();
+    await searchPage.searchOverlay.searchFor(file3);
+
+    await personalFiles.dataTable.selectItem(file3);
+    await personalFiles.acaHeader.shareButton.click();
+
+    const urlBefore = await personalFiles.shareDialog.getLinkUrl();
+    await personalFiles.shareDialog.shareToggle.click();
+    await personalFiles.confirmDialog.cancelButton.click();
+    expect(await personalFiles.shareDialog.isShareToggleChecked()).toBe(true);
+
+    const urlAfter = await personalFiles.shareDialog.getLinkUrl();
+    expect(urlBefore).toEqual(urlAfter);
+  });
+
+  test('[C306999] Unshare a file from the context menu', async ({ personalFiles, searchPage, nodesApiAction, page }) => {
+    await personalFiles.acaHeader.searchButton.click();
+    await searchPage.searchInput.searchButton.click();
+    await searchPage.searchOverlay.checkFilesAndFolders();
+    await searchPage.searchOverlay.searchFor(file4);
+
+    await personalFiles.dataTable.rightClickOnItem(file4);
+    await personalFiles.matMenu.clickMenuItem('Shared Link Settings');
+    const url = await personalFiles.shareDialog.getLinkUrl();
+    await personalFiles.shareDialog.shareToggle.click();
+
+    await personalFiles.confirmDialog.okButton.click();
+    expect(await nodesApiAction.isFileShared(file4Id)).toBe(false);
```

Comment:
we can move after Check on UI as we are getting flaky test due to backend take time to reflect file status.

Reply:
Moved this line after check on UI, Thanks.